plugin/trino-hive/src/main/java/io/trino/plugin/hive/DefaultHiveTableRedirectionsProvider.java

```diff
@@ -18,15 +18,30 @@
 import io.trino.spi.connector.ConnectorSession;
 
 import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
 
+import static io.trino.plugin.hive.HiveSessionProperties.getDeltaLakeCatalogName;
 import static io.trino.plugin.hive.HiveSessionProperties.getIcebergCatalogName;
+import static io.trino.plugin.hive.util.HiveUtil.isDeltaLakeTable;
 import static io.trino.plugin.hive.util.HiveUtil.isIcebergTable;
 
 public class DefaultHiveTableRedirectionsProvider
         implements HiveTableRedirectionsProvider
 {
     @Override
     public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, Table table)
+    {
+        return Stream.<Supplier<Optional<CatalogSchemaTableName>>>of(
+                        () -> redirectTableToIceberg(session, table),
+                        () -> redirectTableToDeltaLake(session, table))
+                .map(Supplier::get)
+                .filter(Optional::isPresent)
+                .findFirst()
```

Comment:
we shouldn't invoke `redirectTableToDeltaLake` if `redirectTableToIceberg` returns non-empty?
does the stream guarantee this? even if it does, this can be written more explicitly:

```java
return redirectTableToIceberg(session, table)
  .or(() -> redirectTableToDeltaLake(session, table));
```

Reply:
Funny how overcomplicated I wrote the first version.
Thank you for the improvement suggestion.